src/types.rs

```diff
@@ -303,34 +338,48 @@ impl Id {
       public_keys : (pub_sign_key, pub_asym_key),
       secret_keys : (sec_sign_key, sec_asym_key),
       validation_token : validation_token,
-      name : NameType::new(digest.0)
+      name : NameType::new(digest.0),
     }
   }
 
-  pub fn get_name(&self) -> NameType {
-    self.name.clone()
+  pub fn get_name<'a>(&'a self) -> &'a NameType {
```

Comment:
Can you return this to returning the actual NameType with implicit clone inside of the function?